### PR TITLE
Add a new analyzer implementation to address the new ADA strategies

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
This PR makes two changes:

- add configuration fields for the new ADA strategies
- add new analyzer implementation so that we can implement for each strategy after this.

This PR doesn't change the code that will be run in any context. Just preparing for the next step.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nakabonne** [comment]
Supplement:

Currently, the analysis executor performs static ADA by just letting an analysis provider evaluate. (see [here](https://github.com/pipe-cd/pipe/blob/814794b442bac526472e3ff5ce392c94ca4b9275/pkg/app/piped/executor/analysis/analyzer.go#L61-L96))

But in this way, it's a bit difficult to compare two variants. So I settled on adding a new analyzer that evaluates itself and looking to replace the above legacy implementation with it.

File: pkg/app/piped/livestatestore/BUILD.bazel [hunk 1]
```diff
@@ -8,7 +8,6 @@ go_library(
     deps = [
         "//pkg/app/piped/cloudprovider/kubernetes:go_default_library",
         "//pkg/app/piped/livestatestore/cloudrun:go_default_library",
-        "//pkg/app/piped/livestatestore/ecs:go_default_library",
```
Comment (khanhtc1202) [inline pkg/app/piped/livestatestore/BUILD.bazel:11]: why do we have this change? 🤔

Comment (nakabonne) [reply pkg/app/piped/livestatestore/BUILD.bazel:11]: For sure. Let me undo it.

**Pull request merged**